Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/StorageModule.kt

```diff
@@ -1,18 +1,56 @@
 package de.rki.coronawarnapp.storage
 
+import android.content.Context
+import androidx.datastore.preferences.SharedPreferencesMigration
+import androidx.datastore.preferences.core.PreferenceDataStoreFactory
+import androidx.datastore.preferences.preferencesDataStoreFile
 import dagger.Binds
 import dagger.Module
+import dagger.Provides
 import dagger.multibindings.IntoSet
+import de.rki.coronawarnapp.util.coroutine.AppScope
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.di.AppContext
 import de.rki.coronawarnapp.util.reset.Resettable
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.plus
+import javax.inject.Qualifier
+import javax.inject.Singleton
 
-@Module
-interface StorageModule {
+@Module(includes = [StorageModule.ResetModule::class])
+object StorageModule {
 
-    @Binds
-    @IntoSet
-    fun bindResettableOnboardingSettings(resettable: OnboardingSettings): Resettable
+    @Singleton
+    @Provides
+    @StorageDataStore
+    fun provideDataStore(
+        @AppContext context: Context,
+        @AppScope appScope: CoroutineScope,
+        dispatcherProvider: DispatcherProvider
+    ) = PreferenceDataStoreFactory.create(
+        scope = appScope + dispatcherProvider.IO,
+        produceFile = { context.preferencesDataStoreFile(STORAGE_DATASTORE_NAME) },
+        migrations = LEGACY_SHARED_PREFS.map { SharedPreferencesMigration(context, it) }
+    )
 
-    @Binds
-    @IntoSet
-    fun bindResettableTracingSettings(resettable: TracingSettings): Resettable
+    @Module
+    internal interface ResetModule {
+        @Binds
+        @IntoSet
+        fun bindResettableStorageReset(resettable: StorageReset): Resettable
+    }
 }
+
+private const val STORAGE_DATASTORE_NAME = "storage_localdata"
```

Comment:
please be specific ? "storage_localdata" is very generic

Reply:
I will make it "storage_settings_localdata" !?

Reply:
Renamed to "shared_settings_storage" because this is a storage which stores settings which are shared between multiple packages

Reply:
See https://github.com/corona-warn-app/cwa-app-android/pull/5236#discussion_r894230617